### PR TITLE
Remove some more ParseFieldMatcher usages

### PR DESCRIPTION
**Pull request by javanna**
Remove some more ParseFieldMatcher usages

Relates to #19552
Relates to #22130

### CODE REVIEW

**Comment by javanna** [comment]
@nik9000 This should be the second last PR on this topic, next one will be the actual removal I think.

File: core/src/main/java/org/elasticsearch/action/admin/indices/alias/IndicesAliasesRequest.java [hunk 1]
```diff
@@ -157,18 +157,17 @@ public static AliasActions removeIndex() {
                 }
             }, new ParseField("filter"));
             // Since we need to support numbers AND strings here we have to use ValueType.INT.
-            ADD_PARSER.declareField(AliasActions::routing, p -> p.text(), new ParseField("routing"), ValueType.INT);
-            ADD_PARSER.declareField(AliasActions::indexRouting, p -> p.text(), new ParseField("index_routing"), ValueType.INT);
-            ADD_PARSER.declareField(AliasActions::searchRouting, p -> p.text(), new ParseField("search_routing"), ValueType.INT);
+            ADD_PARSER.declareField(AliasActions::routing, XContentParser::text, new ParseField("routing"), ValueType.INT);
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/admin/indices/alias/IndicesAliasesRequest.java]: ++

File: core/src/main/java/org/elasticsearch/action/admin/indices/shrink/ShrinkRequest.java [hunk 2]
```diff
@@ -40,14 +39,11 @@
  */
 public class ShrinkRequest extends AcknowledgedRequest<ShrinkRequest> implements IndicesRequest {
 
-    public static final ObjectParser<ShrinkRequest, ParseFieldMatcherSupplier> PARSER =
-        new ObjectParser<>("shrink_request", null);
+    public static final ObjectParser<ShrinkRequest, Void> PARSER = new ObjectParser<>("shrink_request", null);
     static {
-        PARSER.declareField((parser, request, parseFieldMatcherSupplier) ->
-                request.getShrinkIndexRequest().settings(parser.map()),
+        PARSER.declareField((parser, request, parseFieldMatcherSupplier) -> request.getShrinkIndexRequest().settings(parser.map()),
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/action/admin/indices/shrink/ShrinkRequest.java]: s/parseFieldMatcherSupplier/context/ ? Or is there a version of this method without context?

**Review by nik9000 — APPROVED** [review]
Left a variable naming thing but otherwise LGTM.

File: core/src/main/java/org/elasticsearch/action/admin/indices/shrink/ShrinkRequest.java (hunk 2 above)
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/admin/indices/shrink/ShrinkRequest.java]: leftover thanks for finding it.

**Pull request merged**